packages/common/src/projects/HubProject.ts

```diff
@@ -282,7 +282,37 @@ export class HubProject
     delete editor.view.featuredImage;
 
     // convert back to an entity
-    const entity = editorToProject(editor, this.context.portal);
+    let entity = editorToProject(editor, this.context.portal);
+
+    // handle metrics
+    const _metric = editor._metric;
+
+    // if we have a current metric and it's beyond the default empty from toEditor
+    if (_metric && Object.keys(_metric).length) {
+      let metricId = editorContext?.metricId;
+
+      // creating a new metric
+      if (!metricId) {
+        metricId = createId(camelize(`${editor._metric.cardTitle}_`));
+      }
+
+      // transform editor values into metric and displayConfig
+      const { metric, displayConfig } = editorToMetric(
+        editor._metric,
+        metricId,
+        {
+          metricName: editor._metric.cardTitle,
+          entityInfo: {
```

Comment:
If I'm understanding correctly, `getEntityMetrics` only grabs the metrics off of an entity and returns it with preprocessed data (entityInfo)...but doesn't actually save the preprocessed data onto the entity's metrics. 

FromEditor is where we'd save the metric onto the entity, and we'd want the entityInfo to be there -- ._metric only holds the editor values, so entity info isn't there -- if we continued, the saved metric would just have entityInfo as object with undefined name/id/type...it would be filled in when we eventually get it on the front-end, but it would never be persisted

wouldn't we want the entityInfo to actually be persisted with the metric, though?

Reply:
I think I get what you're saying, but yeah, I'm wondering if we actually even need `entityInfo` to be persisted on the metric? When I look at the `IMetric`, the comment above `entityInfo` is: 

```ts
/**
 * Information about the source entity that will be attached to the `IMetricFeature` when it's resolved.
 * This will be added to the `IMetric` as it's processed.
 */
entityInfo?: IEntityInfo;
```

which makes it seem like it's not supposed to be persisted, and is only attached during pre-processing? I guess I'm not certain what the intent behind this property even is, so might be a question for Dave?